crates/consensus/beacon/src/engine/mod.rs

```diff
@@ -799,6 +799,32 @@ where
                             return Some(Ok(()))
                         }
 
+                        // update the canon chain if continuous is enabled
+                        if self.sync.run_pipeline_continuously() {
+                            let max_block = ctrl.progress().unwrap_or_default();
```

Comment:
extracting best block lookup to a separate function will be cleaner in terms of error handlig